cou/utils/cli.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Canonical Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Command line interaction utilities."""
+
+import asyncio
+import logging
+from typing import NoReturn
+
+from halo import Halo
+
+from cou.steps import UpgradeStep
+
+logger = logging.getLogger(__name__)
+
+
+async def _cancel_plan(plan: UpgradeStep, progress_indicator: Halo) -> NoReturn:
```

Comment:
nit: if we instantiate a progress_indicator in this module we don't need to have it as a parameter for this function.

Reply:
What do you mean by that?

Reply:
After discussion we agreed that indicator initialization should be moved to `utils/__init__.py`. Same as we did in #164.